src/MongoDB.Driver.Core/ChangeStreamDocumentSerializer.cs

```diff
@@ -47,149 +39,21 @@ public class ChangeStreamDocumentSerializer<TDocument> : SealedClassSerializerBa
-            ChangeStreamOperationType? operationType = null;
-            ChangeStreamUpdateDescription updateDescription = null;
-
-            reader.ReadStartDocument();
-            while (reader.ReadBsonType() != 0)
-            {
-                var fieldName = reader.ReadName();
-                switch (fieldName)
-                {
-                    case "_id":
-                        resumeToken = BsonDocumentSerializer.Instance.Deserialize(context);
-                        break;
-
-                    case "ns":
-                        collectionNamespace = DeserializeCollectionNamespace(reader);
-                        break;
-
-                    case "documentKey":
-                        documentKey = BsonDocumentSerializer.Instance.Deserialize(context);
-                        break;
-
-                    case "fullDocument":
-                        if (reader.CurrentBsonType == BsonType.Null)
-                        {
-                            reader.ReadNull();
-                            fullDocument = default(TDocument);
-                        }
-                        else
-                        {
-                            fullDocument = _documentSerializer.Deserialize(context);
-                        }
-                        break;
-
-                    case "operationType":
-                        operationType = __operationTypeSerializer.Deserialize(context);
-                        break;
-
-                    case "updateDescription":
-                        updateDescription = __updateDescriptionSerializer.Deserialize(context);
-                        break;
 
-                    case "clusterTime":
-                        // TODO: decide what to do about clusterTime
-                        reader.SkipValue();
-                        break;
-
-                    default:
-                        throw new FormatException($"Invalid field name: \"{fieldName}\".");
-                }
-            }
-            reader.ReadEndDocument();
-
-            return new ChangeStreamDocument<TDocument>(
-                resumeToken,
-                operationType.Value,
-                collectionNamespace,
-                documentKey,
-                updateDescription,
-                fullDocument);
+            RegisterMember("ClusterTime", "clusterTime", BsonDocumentSerializer.Instance);
+            RegisterMember("CollectionNamespace", "ns", ChangeStreamDocumentCollectionNamespaceSerializer.Instance);
+            RegisterMember("DocumentKey", "documentKey", BsonDocumentSerializer.Instance);
+            RegisterMember("FullDocument", "fullDocument", _documentSerializer);
+            RegisterMember("OperationType", "operationType", ChangeStreamOperationTypeSerializer.Instance);
+            RegisterMember("ResumeToken", "_id", BsonDocumentSerializer.Instance);
+            RegisterMember("UpdateDescription", "updateDescription", ChangeStreamUpdateDescriptionSerializer.Instance);
```

Comment:
All we have to do is register the members. The base class takes care of the rest.